tests/unit/test_endpoint_provider.py

```diff
@@ -544,3 +553,141 @@ def test_deeply_nested(self):
             ]
         }
         self.assertDictEqual(expected_rules_file, rules_file_dict_read)
+
+
+def provider_with_rules_factory(rules: Dict[str, str]):
+    """Class factory to create a provider charm class with the given rules.
+
+    Args:
+        rules: a mapping between filename and contents.
+    """
+
+    class ProviderCharm(CharmBase):
+        metadata_yaml = textwrap.dedent(
+            """
+            provides:
+              metrics-endpoint:
+                interface: prometheus_scrape
+            """
+        )
+
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args)
+
+            self.sandbox = TempFolderSandbox()
+            alert_rules_path = os.path.join(self.sandbox.root, "alert_rules")
+            for filepath, contents in rules.items():
+                self.sandbox.put_file(os.path.join(alert_rules_path, filepath), contents)
+
+            self.provider = MetricsEndpointProvider(
+                self, jobs=JOBS, alert_rules_path=alert_rules_path
+            )
+
+    return ProviderCharm
+
+
+class TestAlertRulesDoNotContainUnitNumber(unittest.TestCase):
+    """Feature: Consumer lib must not add a unit number to rule filters or labels.
+
+    Background: In lieu of SDI or pydantic, schema verification is done in unit tests.
```

Comment:
I do not see the significance of this comment. As I understand our tests should not be trying to validate the "schema" of the alert rule. The tests instead need to validate to alternative conditions
- If `juju_unit` is present in an alert label or expression it is not removed
- If `juju_unit` is not present in an alert label or expression it is not added.

This has nothing to do with "schema verification" i.e. we are not trying to validate the syntax of alert rules, their expressions or their labels.